src/main/java/com/facebook/openwifirrm/modules/ProvMonitor.java

```diff
@@ -105,25 +110,41 @@ private void runImpl() {
 		// ignoring "entity" completely
 		InventoryTagList inventory = client.getProvInventory();
 		SerialNumberList inventoryForRRM = client.getProvInventoryForRRM();
-		VenueList venueList = client.getProvVenues();
 		//EntityList entityList = client.getProvEntities();
 		if (inventory == null || inventoryForRRM == null) {
 			logger.error("Failed to fetch inventory from owprov");
 			return;
 		}
+
+		// fetch the RRM details for each AP that has RRM enabled
+		Map<String, RRMDetails> rrmDetails = new HashMap<String, RRMDetails>();
+		for (String serialNumber : inventoryForRRM.serialNumbers) {
+			RRMDetails details =
+				client.getProvInventoryRrmDetails(serialNumber);
+			if (details == null) {
+				logger
+					.error("Could not fetch RRM details for {}", serialNumber);
+				continue;
+			}
+
+			rrmDetails.put(serialNumber, details);
+		}
```

Comment:
As discussed, we should assume that RRM config is only set at venue level, and the workaround with current Prov APIs would be to query `RRMDetails` for one (random/first) device in each venue -- you don't need to query all devices. And hopefully `getProvInventoryForRRM()` API still works here to filter RRM-enabled devices only (should test this if you haven't already).